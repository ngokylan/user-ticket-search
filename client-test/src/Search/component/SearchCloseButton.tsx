import React from 'react';
import Button from '../../Button';
import { Text } from '../../Typography';

export type SearchCloseButtonProps = {
  onClick: () => void;
};

function SearchCloseButton({ onClick }: SearchCloseButtonProps) {
  return (
    <div className="search__close">
      <Button isText={true} onClick={onClick}>
        <Text size="button">Cancel</Text>
      </Button>
    </div>
  );
}

export default SearchCloseButton;
