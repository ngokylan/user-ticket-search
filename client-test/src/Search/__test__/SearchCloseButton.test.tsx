import React from 'react';
import { mount } from 'enzyme';
import SearchCloseButton, {
  SearchClearButtonProps,
} from '../component/SearchCloseButton';

const defaultProps: SearchClearButtonProps = undefined;
const mountComponent = (customProps: any = {}) => {
  const props = { ...defaultProps, ...customProps };
  const wrapper = mount(<SearchCloseButton {...props} />);
  const component = wrapper.find('.search__close');

  return { props, wrapper, component };
};

describe('Search::SearchClearButton', () => {
  describe('rendering', () => {
    it('should render component without any crash', () => {
      const { wrapper } = mountComponent({ value: 'some-value' });
      expect(wrapper).toHaveLength(1);
    });

    it('should be match with snapshot', () => {
      const { wrapper } = mountComponent();
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('props', () => {
    describe('default', () => {
      it('should have expected default props', () => {
        expect(JSON.stringify(defaultProps)).toBe(
          JSON.stringify(SearchCloseButton.defaultProps)
        );
      });
    });

    describe('children', () => {
      it('should render children', () => {
        const text = 'Cancel';
        const { wrapper } = mountComponent();
        expect(wrapper.find('Text').text()).toBe(text);
      });
    });

    describe('custom', () => {
      it.each([['className', 'search__close', {}]])(
        'should render prop %s as %s',
        (name, expected, props) => {
          const { component } = mountComponent(props);
          expect(component.prop(name)).toBe(expected);
        }
      );

      it.each([['size', 'button', {}]])(
        'should render prop %s as %s',
        (name, expected, props) => {
          const { component } = mountComponent(props);
          expect(component.find('Text').prop(name)).toBe(expected);
        }
      );
    });
  });

  describe('interaction', () => {
    it.each([
      ['should', 'Button', 1, { value: 'some-value' }],
      ['should', 'Button', 1, {}],
      ['should', 'Text', 1, {}],
    ])('%s have %s', (name, element, expected, props) => {
      const { wrapper } = mountComponent(props);
      expect(wrapper.find(element)).toHaveLength(expected);
    });
  });
});
