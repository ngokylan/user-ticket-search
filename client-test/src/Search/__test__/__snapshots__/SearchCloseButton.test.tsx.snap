// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search::SearchClearButton rendering should be match with snapshot 1`] = `
<div
  className="search__close"
>
  <button
    className="sc-bdfBwQ gjsWpT btn btn--default btn--is-uppercase btn--is-text"
    data-testid="btn-default"
    onBlur={[Function]}
    onClick={[Function]}
    role="button"
    type="button"
  >
    <span
      className="btn__label"
      data-testid="btn-label-default"
    >
      <span
        className="sc-iBPRYJ typography text-size-button"
        data-testid="typography-default"
      >
        Cancel
      </span>
    </span>
    <span
      className="btn__hover"
    />
    <span
      className="btn__focused"
    />
    <span
      className="btn__pressed"
    />
  </button>
</div>
`;
